src/main/java/com/ukubuka/core/parser/impl/UkubukaXMLParser.java

```diff
@@ -1,23 +1,35 @@
 package com.ukubuka.core.parser.impl;
 
-import java.util.Map;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.*;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
 
 import com.ukubuka.core.exception.ParserException;
+import com.ukubuka.core.exception.ReaderException;
+import com.ukubuka.core.model.ExtractFlags;
 import com.ukubuka.core.model.FileContents;
+import com.ukubuka.core.model.SupportedSource;
+import com.ukubuka.core.parser.UkubukaBaseParser;
 import com.ukubuka.core.parser.UkubukaParser;
+import com.ukubuka.core.utilities.Constants;
+import com.ukubuka.core.utilities.Utilities;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamReader;
 
 /**
  * Ukubuka XML Parser
  * 
- * @author agrawroh
+ * @author aswinv
  * @version v1.0
  */
 @Component("UkubukaXMLParser")
-public class UkubukaXMLParser implements UkubukaParser {
+public class UkubukaXMLParser extends UkubukaBaseParser implements UkubukaParser {
 
     /************************************ Logger Instance ***********************************/
     private static final Logger LOGGER = LoggerFactory
```

Comment:
Why do we still use `UkubukaDFileParser` here and not this class? It won't log any events from this class correctly!

Reply:
Fixed